### PR TITLE
tests(smoke): deep clone expectations. do not use p.stdout

### PR DESCRIPTION
**Pull request by connorjclark**
Found two issues when trying to consume this internally.

1) we clone expectations out of an abundance of caution, but we lose regexps currently. need a regexp replacer/reviver
2) process.stdout doesn't play nice with how I need to use it. just replaced it with a simple console.log.

### CODE REVIEW

**Comment by exterkamp** [comment]
@patrickhulce I choose you!

File: lighthouse-cli/test/smokehouse/frontends/lib.js [hunk 1]
```diff
@@ -16,14 +16,48 @@
...
+
 /**
  * @param {Smokehouse.SmokehouseLibOptions} options
  */
 async function smokehouse(options) {
   const {urlFilterRegex, skip, modify, ...smokehouseOptions} = options;
 
   /** @type {Smokehouse.TestDfn[]} */
-  const clonedTests = JSON.parse(JSON.stringify(smokeTests));
+  const clonedTests = deepCopy(smokeTests);
```
Comment (patrickhulce) [inline lighthouse-cli/test/smokehouse/frontends/lib.js]: this is just dev dependency why not use just use `_.cloneDeep`? :) https://lodash.com/docs/4.17.15#cloneDeep

Comment (patrickhulce) [reply lighthouse-cli/test/smokehouse/frontends/lib.js]: if we decide to keep and maintain this ourselves instead I expect rigorous tests and edge cases ;)

Comment (connorjclark) [reply lighthouse-cli/test/smokehouse/frontends/lib.js]: sgtm

**Comment by connorjclark** [comment]
@paulirish what up with pr lint
